src/hotspot/share/nmt/nmtPreInit.hpp

```diff
@@ -267,7 +267,6 @@ class NMTPreInit : public AllStatic {
   // Returns true if allocation was handled here; in that case,
   // *rc contains the return address.
   static bool handle_malloc(void** rc, size_t size) {
-    size = MAX2((size_t)1, size);         // malloc(0)
```

Comment:
Not this change, but why are these functions in the hpp file? They seem more appropriate in the cpp file.